### ISSUE TITLE
Unnecessary amount of structures

### ISSUE THREAD

**Issue by kannagikazuko** [campbellbrendene/discord-command-parser#12]
![image](https://media.giphy.com/media/I7kkegrRyNrk4/giphy.gif)

For example, whether a command has been successfully parsed or not can be merged with the state as a simple variable. Data can change accordingly without the need for so many extra definitions. Perhaps this is a pitfall of using TypeScript for something like this...

**Comment by tbnritzdoge** [comment]
I think the developers should rewrite it in coffeescript for maximum efficiency

**Comment by wzhouwzhou** [comment]
@tbnritzdoge How about using Java/dealing with the JVM overhead and having an io bridge, i.e., what Lavalink uses?

**Comment by tbnritzdoge** [comment]
i think they should use assemblyscript obviously

**Comment by wzhouwzhou** [comment]
That is true, NodeJS has more support for WASM and WAT syntax now, so it would be good assuming that they don't use any FFIs...

**Comment by dicedtomatoreal** [comment]
I think we should all be using english?

**Comment by tbnritzdoge** [comment]
i think they should use neon because it go brr

**Comment by wzhouwzhou** [comment]
@campbellbrendene any thoughts on this?
